README.md

```diff
@@ -539,6 +540,30 @@ import { batchRequests } from 'graphql-request';
 })().catch((error) => console.error(error))
 ```
 
+### Cancellation
+
+It is possible to cancel a request using an `AbortController` signal.
+
+You can define the `signal` in the `GraphQLClient` constructor:
+
+```ts
+  const abortController = new AbortController()
+
+  const client = new GraphQLClient(endpoint, { signal: abortController.signal })
+  client.request(query)
+
+  abortController.abort()
+```
+
+You can also set the signal per request (this will override an existing GraphQLClient signal):
+
+```ts
+  const abortController = new AbortController()
+
+  request(endpoint, query, undefined, undefined, abortController.signal)
```

Comment:
I'm not sure how usable this API is. Maybe it should only be possible in object form.

Reply:
Do you mean with a `GraphQLClient` only?

I'm not a super fan of `...,undefined, undefined,...` but we can also add an override for the function `request` with an object as a single argument (more flexible):
`request ({ url, document, variables, headers, signal })`

Reply:
Yeah also thinking of an overload here!

Reply:
@jasonkuhrt I can handle this if you want :smile:

Overloads to add to:
- [ ] `request`
- [ ] `rawRequest`
- [ ]  `batchRequests`

It is fine with you?

Reply:
@arnaudbzn Another PR is fine if we agree that the non-object signature makes sense. I'm actually not sure it does though. WDTY? Do we actually need both signatures to be equally powerful? Or should we treat the non-object one as  just sugar for common cases. That's what I'm inclined towards right now.

Reply:
Agreed to keep the current function signature simple and offer the new signal feature in new overloads using an object.

Reply:
@arnaudbzn Yep let's do that!

Reply:
@jasonkuhrt Changes done. 
- Only the `GraphQLClient` methods `request`, `rawRequest`, `batchRequests`) now have overloads to support a new `option` single argument.
- Added new types `RequestOptions`, `BatchRequestOptions`, `RawRequestOptions`

We can also add overloads to `request`, `rawRequest`, `batchRequests` without GraphQLClient. In this case additional request options types should be added to support the `url` option. WDTY?

Reply:
@arnaudbzn yep sounds good

Reply:
@jasonkuhrt Done! 🙂
- [x] new request options types
- [x] overloads
- [x]  tests
- [x] readme

New types naming (e.g: `RequestOptions` and `RequestExtendedOptions`) could be improved if you have other suggestions :thinking: